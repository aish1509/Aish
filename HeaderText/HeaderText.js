import React from 'react';
import PropTypes from 'prop-types';
import {withStyles} from '@material-ui/core/styles';

const styles = () => ({
	header: {
		marginBlockEnd: 0,
		marginBlockStart: '-3px',
		textTransform: 'uppercase',
		paddingTop: '8px'
	},
	subheader: {
		marginBlockEnd: 0,
		marginBlockStart: '-13px',
		textTransform: 'none',
		paddingTop: '5px'
	}
});

const HeaderText = ({children, classes, subheaderText}) => (
	<header>
		{subheaderText && (
			<h6 className={classes.subheader}>{subheaderText}</h6>
		)}
		<h3 className={classes.header}>{children}</h3>
	</header>
);

HeaderText.defaultProps = {
	classes: {},
	subheaderText: null,
	children: ''
};

HeaderText.propTypes = {
	children: PropTypes.node,
	classes: PropTypes.object,
	subheaderText: PropTypes.string
};

export default withStyles(styles)(HeaderText);
