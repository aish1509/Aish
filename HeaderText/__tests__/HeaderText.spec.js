import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import HeaderText from '../HeaderText';

describe('<HeaderText />', () => {
	let wrapper = null;

	beforeEach(() => {
		wrapper = shallow(<HeaderText>mock header text</HeaderText>);
	});

	it('should render correctly with default props', () => {
		expect(shallowToJson(wrapper.dive())).toMatchSnapshot();
	});

	it('should render correctly with subHeaderText set', () => {
		wrapper.setProps({
			subheaderText: 'mockSubheaderText'
		});
		expect(shallowToJson(wrapper.dive())).toMatchSnapshot();
	});
});
