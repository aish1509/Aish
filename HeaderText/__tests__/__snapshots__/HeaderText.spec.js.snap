// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderText /> should render correctly with default props 1`] = `
<header>
  <h3
    className="HeaderText-header-1"
  >
    mock header text
  </h3>
</header>
`;

exports[`<HeaderText /> should render correctly with subHeaderText set 1`] = `
<header>
  <h6
    className="HeaderText-subheader-2"
  >
    mockSubheaderText
  </h6>
  <h3
    className="HeaderText-header-1"
  >
    mock header text
  </h3>
</header>
`;
